manual/manual/cmds/instrumented-runtime.etex

```diff
@@ -0,0 +1,148 @@
+to {\em pause} events collection after the program has started.
+
+\begin{verbatim}
+        OCAML_EVENTLOG_ENABLED=p ./program
+\end{verbatim}
+
+The program will have to resume events collection explicitly.
+This can be done via {\tt Gc.eventlog_resume} and
+{\tt Gc.eventlog_pause}, available in the {\stdmoduleref{Gc}} module.
+
+\subsection{ss:instr-runtime-limitations}{Limitations}
+
+The instrumented runtime does not support the {\tt fork} system call.
+A child process forked from an instrumented program will not be traced.
+
+The instrumented runtime aims to provide insight into the runtime's execution
+while maintaining a low overhead.
+However, the impact on performance may increase based on the pressure put on the
+garbage collector.
+
+For a program with an extended running time where the collection of only a
+small sample of events is required, using the {\em eventlog_resume} and
+{\em eventlog_pause} primitives may help relieve some of the
+tracing induced performance impact.
+
+\section{s:instr-runtime-read}{Read traces}
+
+Traces generated by the instrumented runtime can be analyzed with tooling
+available outside of the OCaml distribution.
+
+A complete trace is comprised of a {\em metadata file} and a {\em trace file}.
+Two simple ways to work with the traces are the {\em eventlog-tools} and
+{\em babeltrace} libraries.
+
+\subsection{ss:instr-runtime-tools}{eventlog-tools}
+{\em eventlog-tools} is a set of libraries and tools written in OCaml
+that allow to decode and perform basic format conversions and analysis.
+
+For more information about {\em eventlog-tools}, refer to the project's
+main page:
+\ifouthtml
+\ahref{https://github.com/ocaml-multicore/eventlog-tools}
+      {https://github.com/ocaml-multicore/eventlog-tools}.
+\else
+{\tt https://github.com/ocaml-multicore/eventlog-tools}
+\fi
+
+\subsection{ss:instr-runtime-babeltrace}{babeltrace}
+
+{\em babeltrace} is a C library, as well as a Python binding and set of tools
+that serve as the reference implementation for the {\em Common Trace Format}.
+The {\em babeltrace} command line utility allows for a basic rendering
+of a trace's content, while the high level Python API can be used to
+decode the trace and process them programmatically with libraries
+such as {\em numpy} or {\em Jupyter}.
+
+Unlike {\em eventlog-tools}, which possesses a specific knowledge of
+OCaml's {\em Common Trace Format} schema, it is required to provide
+the OCaml {\em metadata} file to {\em babeltrace}.
+
+The metadata file is available in the OCaml installation.
+Its location can be obtained using the following command:
+
+\begin{verbatim}
+        ocamlc -where
+\end{verbatim}
+
+The {\em eventlog_metadata} file can be found at this path and
+copied in the same directory as the generated trace file.
+However, {\em babeltrace} expects the file to be named
+{\tt metadata} in order to process the trace.
+Thus, it will need to be renamed when copied to the trace's directory.
```

Comment:
Again, showing a concrete example of that (the shell sequence that does a copy and then invokes babeltrace to do *somthing*) would be useful.

Reply:
Noted for both, I will provide concrete examples, thanks.